ckanext/xroad_integration/harvesters/xroad_harvester.py

```diff
@@ -366,41 +380,40 @@ def import_stage(self, harvest_object):
                 'access_restriction_level': 'public'
             }
 
-            if service_description and service_description.data:
-                service_description_data_utf8 = service_description.data.encode('utf-8')
+            if service_description_data is not None:
+                service_description_data_utf8 = service_description_data.encode('utf-8')
 
-                # Todo: Validity of openapi ?
-                if service.wsdl:
-                    valid_wsdl = self._is_valid_wsdl(service_description_data_utf8)
+                if service_description_type == 'wsdl':
+                    valid_content = self._is_valid_wsdl(service_description_data_utf8)
                     timestamp_field = 'wsdl_timestamp'
                     target_name = 'service.wsdl'
                     resource_format = 'wsdl'
-                else:
-                    valid_wsdl = True
+                elif service_description_type == 'openapi':
+                    # TODO: Validity of openapi ?
+                    valid_content = True
                     timestamp_field = 'openapi_timestamp'
                     target_name = 'service.json'
                     resource_format = 'openapi-json'
+                elif service_description_type == 'rest':
+                    valid_content = True
+                    timestamp_field = 'rest_timestamp'
+                    target_name = 'service.json'
+                    resource_format = 'openapi-json'
```

Comment:
UI should differ these from openapi descriptions. Would require changes to https://github.com/vrk-kpa/ckanext-openapiviewer/blob/8e6873bc1d8478655e82d5ec95db5c6d6ab532c2/ckanext/openapi/plugin.py#L17 to support some other format, for example just list it as rest